### ISSUE TITLE
Do not call hooks on cron mode CLI. Correct on mode GLPI 9.4 and 9.4.1.1

### ISSUE THREAD

**Issue by supervisando** [glpi-project/glpi#5624]
When cron is in mode CLI don't call hook, but in mode GLPI work

I need hook for mail to close ticket when i receive a special mail 

I put the example of plugin don't work 
**setup.php**
```php
<?php
function plugin_init_workwithmail() {
   global $PLUGIN_HOOKS,$CFG_GLPI;
   $PLUGIN_HOOKS['csrf_compliant']['workwithmail'] = true;
   $PLUGIN_HOOKS['pre_item_add']['workwithmail']= array(
      'Ticket'       => 'plugin_workwithmail',
      'TicketFollowup'  => 'plugin_workwithmail',
	  'ITILFollowup'	=> 'plugin_workwithmail'
   );
   $PLUGIN_HOOKS['item_add']['workwithmail']= array(
      'Ticket'       => 'plugin_workwithmail',
      'TicketFollowup'  => 'plugin_workwithmail',
	  'ITILFollowup'	=> 'plugin_workwithmail'
   );
}

function plugin_version_workwithmail() {
   return array('name'           => 'Example of problem',
                'version'        => '1.1',
                'author'         => 'I am the author',
                'minGlpiVersion' => '0.94');
}

function plugin_workwithmail_check_prerequisites() {
   return true;
}

function plugin_workwithmail_check_config($verbose=false) {
    return true;
}
?>
```
**hook.php**
```php
<?php
function plugin_workwithmail($item) {
	Toolbox::logInFile("WorkWithMail","This work with cron GLPI but not with CLI".PHP_EOL);
	/*
		My code
	*/
}

function plugin_workwithmail_install() {
   return true;
}

function plugin_workwithmail_uninstall() {
   return true;
}
?>
```

**Comment by cedric-anne** [comment]
Hi,

Are you using APCu or Wincache ?

If you are not using them, can you try to empty the `files/_cache` directory, check if the plugin is correctly enabled and run the cron again ?

Regards

**Comment by supervisando** [comment]
I use APCu
OS: Red Hat Enterprise Linux Server release 7.6
php: PHP 7.3.1
The cron is owned by apache.
The only "special" configuration is files directory. It's a link to other path, but i don't have problem with this.

The plugin is enable, and work if lunch web task or it is mode GLPI.
The problem isn't the linux cron. When i lunch php cron.php on shell don't work.

Thanks.

Linux shell:
-bash-4.2$ _whoami_
apache
-bash-4.2$ _ll_
total 0
-rw-r--r--. 1 apache apache 0 mar 18 10:37 cron.log
-rw-r--r--. 1 apache apache 0 mar 18 09:55 event.log
-rw-r--r--. 1 apache apache 0 mar 18 10:32 mail-debug.log
-rw-r--r--. 1 apache apache 0 mar 18 10:32 mail.log
-rw-r--r--. 1 apache apache 0 mar 18 09:55 php-errors.log
-rw-r--r--. 1 apache apache 0 mar 18 09:39 sql-errors.log
Send 2 email to system 
-bash-4.2$ _/usr/bin/php /my.glpi.es/front/cron.php_
...... log mail collect
-bash-4.2$ _ll_
total 44
-rw-r--r--. 1 apache apache   222 mar 18 10:41 cron.log
-rw-r--r--. 1 apache apache     0 mar 18 09:55 event.log
-rw-r--r--. 1 apache apache 33644 mar 18 10:41 mail-debug.log
-rw-r--r--. 1 apache apache   656 mar 18 10:41 mail.log
-rw-r--r--. 1 apache apache     0 mar 18 09:55 php-errors.log
-rw-r--r--. 1 apache apache     0 mar 18 09:39 sql-errors.log
2 email. Open correctly ticket, but don't call hook plugin

Clear all files
-bash-4.2$ _rm *_

Send 2 email to system 
Launch task in web
-bash-4.2$ _ll_
total 32
-rw-r--r--. 1 apache apache   152 mar 18 10:44 cron.log
-rw-r--r--. 1 apache apache 17104 mar 18 10:44 mail-debug.log
-rw-r--r--. 1 apache apache   332 mar 18 10:44 mail.log
**-rw-r--r--. 1 apache apache   176 mar 18 10:44 WorkWithMail.log**
-bash-4.2$ _more WorkWithMail.log_
2019-03-18 09:44:26 [98@my.glpi.es]
This work with cron GLPI but not with CLI
2019-03-18 09:44:27 [98@my.glpi.es]
This work with cron GLPI but not with CLI

**Comment by cedric-anne** [comment]
I cannot reproduce. Maybe it is a cache issue. Could you test to apply #5611, then check that the plugin is still considered as active in the plugins administration page, and do a test ?

**Comment by cedric-anne** [comment]
If `apc.enable` is set to `true` but `apc.enable_cli` is not, then #5629 should fix this.

Regards

**Comment by supervisando** [comment]
Ok. All correct.
When I change and put apc.enable_cli=1 work or if apply #5629 fix my problem

Thanks

**Comment by cedric-anne** [comment]
Great !

Thanks for the feedback.

Regards

**Comment by cedric-anne** [comment]
Fixed by #5629.
